project/frontend/src/components/RestaurantCard.js

```diff
@@ -17,49 +17,106 @@ import Clock from '../assets/clock.svg';
  *     information.
  * @param {!Object<string, *>} props.details An object containing detailed restaurant
  *     information.
+ * @param {?Object<string, *>} props.style An optional style object to be applied
+ *     to the parent container of this component.
+ * @param {boolean=} props.collapsed An optional control over whether this card is in
+ *     a collapsed or full view (default: false).
  */
 function RestaurantCard(props) {
-  const { restaurant, details } = props;
+  const { restaurant, details, style, collapsed = false } = props;
 
-  const photoUrls = details.result.photos.map(
-    (photo) =>
-      'https://maps.googleapis.com/maps/api/place/photo?photoreference=' +
-      photo.photo_reference +
-      '&key=' +
-      process.env.REACT_APP_GOOGLE_API_KEY +
-      '&maxwidth=344'
-  );
+  const photoUrls = details
+    ? details.result.photos.map(
+        (photo) =>
+          `https://maps.googleapis.com/maps/api/place/photo?photoreference=${photo.photo_reference}&key=${process.env.REACT_APP_GOOGLE_API_KEY}&maxwidth=344`
+      )
+    : ['https://pixelpapa.com/wp-content/uploads/2018/11/26.gif'];
+
+  const collapsible = useRef(null);
+  const collapsibleMaxHeight = useRef(null);
+  const restaurantName = useRef(null);
+
+  useLayoutEffect(() => {
+    collapsibleMaxHeight.current = collapsible.current.offsetHeight;
+    if (collapsed) {
+      collapsible.current.style.height = '0px';
+      restaurantName.current.classList.add('collapsed');
+    } else {
+      collapsible.current.style.height = collapsibleMaxHeight.current + 'px';
+    }
+    // We only want this effect to run once (the very first time this component
+    // is laid out) so that we may capture the uncollapsed height of the collapsible
+    // div within this card and save it to a persistent ref (collapsibleMaxHeight).
```

Comment:
This seems hacky, not sure if I'm fully understanding this. If we just want this to capture the uncollapsed height of the card, why do we have the if else?

Reply:
It's necessary to set the collapsible div to its true height according to the `collapsed` prop. On the first render of the component the collapsible div will be rendered at full height regardless of whether `collapsed` is true or false, so that height can be captured in this `useLayoutEffect` and then the height is updated to reflect the `collapsed` state right before the component is painted to the screen. `useLayoutEffect` runs right before components are painted.

Reply:
The overall reason for this method is that the height of the collapsible div will be different for different restaurant cards, and for it to be animated smoothly between collapsed and uncollapsed states, I need an explicit `px` value target. Going from `0px` to `auto` or 'null' will not animate, it will just instantly render at full height. This is a common problem faced with css transitions, and from what I've seen, calculating and saving the height of a variable height component is a common solution to the issue.